wurst/_handles/Destructable.wurst

```diff
@@ -69,6 +69,32 @@ public function destructable.setOccluderHeight(real height)
 public function destructable.getName() returns string
 	return GetDestructableName(this)
 
+// Extensions
+/** used to toggles a destructable gate open close state*/
+public function destructable.gateToggle()
+	if GetDestructableLife(this) > 0
+		ModifyGateBJ(bj_GATEOPERATION_CLOSE, this)
```

Comment:
`ModifyGateBJ` is a BJ function and thus can be implemented directly in wurst.
If done so, it also would not need bj_GATEOPERATION_# "enum" to "switch" between different actions